sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/options/DataLakeFileFlushOptions.java

```diff
@@ -0,0 +1,203 @@
+     * @return the updated DataLakeFileFlushOptions object.
+     */
+    public DataLakeFileFlushOptions setPathHttpHeaders(PathHttpHeaders pathHttpHeaders) {
+        this.pathHttpHeaders = pathHttpHeaders;
+        return this;
+    }
+
+    /**
+     * Optional {@link DataLakeRequestConditions} conditions that are set on the flush of this file.
+     *
+     * @return {@link DataLakeRequestConditions} for this file.
+     */
+    public DataLakeRequestConditions getRequestConditions() {
+        return requestConditions;
+    }
+
+    /**
+     * Optional {@link DataLakeRequestConditions} conditions to add on the flush of this file.
+     *
+     * @param requestConditions {@link DataLakeRequestConditions} to set on this file.
+     * @return the updated DataLakeFileFlushOptions object.
+     */
+    public DataLakeFileFlushOptions setRequestConditions(DataLakeRequestConditions requestConditions) {
+        this.requestConditions = requestConditions;
+        return this;
+    }
+
+    /**
+     * Get lease action set on file.
+     * {@link LeaseAction#ACQUIRE} will attempt to acquire a new lease on the file,
+     * with {@link DataLakeFileFlushOptions#proposedLeaseId} as the lease ID.
+     * {@link LeaseAction#ACQUIRE_RELEASE} will attempt to aquire a new lease on the file,
+     * with {@link DataLakeFileFlushOptions#proposedLeaseId} as the lease ID. The lease will be released once the
+     * Append operation is complete.
+     * {@link LeaseAction#AUTO_RENEW} will attempt to renew the lease specified by {@link DataLakeRequestConditions#getLeaseId()}.
+     * {@link LeaseAction#RELEASE} will attempt to release the least specified by {@link DataLakeRequestConditions#getLeaseId()}.
+     *
+     * @return The {@link LeaseAction} set on the file.
+     */
+    public LeaseAction getLeaseAction() {
+        return leaseAction;
+    }
+
+    /**
+     * Get lease action set on file.
+     * {@link LeaseAction#ACQUIRE} will attempt to acquire a new lease on the file,
+     * with {@link DataLakeFileFlushOptions#proposedLeaseId} as the lease ID.
+     * {@link LeaseAction#ACQUIRE_RELEASE} will attempt to aquire a new lease on the file,
+     * with {@link DataLakeFileFlushOptions#proposedLeaseId} as the lease ID. The lease will be released once the
+     * Append operation is complete.
+     * {@link LeaseAction#AUTO_RENEW} will attempt to renew the lease specified by {@link DataLakeRequestConditions#getLeaseId()}.
+     * {@link LeaseAction#RELEASE} will attempt to release the least specified by {@link DataLakeRequestConditions#getLeaseId()}.
+     *
+     * @param leaseAction the {@link LeaseAction} to set on the file.
+     * @return the updated DataLakeFileFlushOptions object.
+     */
+    public DataLakeFileFlushOptions setLeaseAction(LeaseAction leaseAction) {
+        this.leaseAction = leaseAction;
+        return this;
+    }
+
+    /**
+     * @return the lease duration in seconds.
+     */
+    public Integer getLeaseDuration() {
```

Comment:
Same comment around using `Duration` instead of a number

Reply:
See my comment above.